crates/db_schema/src/source/comment.rs

```diff
@@ -141,18 +141,33 @@ pub struct CommentActions {
   pub vote_is_upvote: Option<bool>,
 }
 
-#[derive(Clone, derive_new::new)]
+#[derive(Clone)]
 #[cfg_attr(
   feature = "full",
   derive(Insertable, AsChangeset, Serialize, Deserialize)
 )]
 #[cfg_attr(feature = "full", diesel(table_name = comment_actions))]
 pub struct CommentLikeForm {
-  pub person_id: PersonId,
-  pub comment_id: CommentId,
-  pub vote_is_upvote: bool,
-  #[new(value = "Utc::now()")]
-  pub voted_at: DateTime<Utc>,
+  person_id: PersonId,
+  comment_id: CommentId,
+  vote_is_upvote: Option<Option<bool>>,
+  voted_at: Option<Option<DateTime<Utc>>>,
+}
+
+impl CommentLikeForm {
+  pub fn new(comment_id: CommentId, person_id: PersonId, is_upvote: Option<bool>) -> Self {
```

Comment:
Its also not really clear that a `None` means remove vote, since in diesel `None` usually means just don't update, while `Some(None)` means set to null.

I'd either add a doc comment about this exception, or make `is_upvote: Option<Option<bool>>`